configs/defs/elasticsearch.textproto

```diff
@@ -0,0 +1,733 @@
+        "specify the user under which Elasticsearch should run.\n"
+        "2. Restart the Elasticsearch service for the changes to take effect."
+      cis_benchmark: {
+        profile_level: 1
+        severity: LOW
+      }
+      scan_instructions:
+        "generic{check_alternatives:{"
+        "    file_checks:{"
+        "      files_to_check:{"
+        "        process_path:{"
+        "          proc_name:\"java\""
+        "          file_name:\"cmdline\""
+        "          cli_arg_regex:\".*org\\\\.elasticsearch\\\\.server.*\""
+        "        }"
+        "      }"
+        "      permission:{"
+        "        user: {name: \"root\" should_own: false}"
+        "      }"
+        "    }"
+        "}}"
+    }
+  }
+  
+  benchmark_configs: {
+    id: "elasticsearch-logs-permissions"
+    compliance_note: {
+      version: { version: "1.0.0" benchmark_document: "" }
+      title: "Ensure ElasticSearch Logs has Correct Permissions"
+      description:
+        "This check verifies that the Elasticsearch log files have the permissions set to 600. "
+        "The log files contain sensitive information, including error messages, stack traces, "
+        "and other diagnostic details, and it is crucial to protect them from unauthorized access."
+      rationale:
+        "Setting the correct permissions for Elasticsearch log files is important for maintaining "
+        "the security and confidentiality of log data. By ensuring that the log files have permissions "
+        "set to 600, you restrict access to only the owner of the files. This prevents unauthorized "
+        "users from viewing or tampering with the log files, protecting sensitive information and "
+        "maintaining the integrity of the logging system."
+      remediation:
+        "To remediate this issue, follow these steps:\n"
+        "1. Identify the location of the Elasticsearch log files in your environment.\n"
+        "   - The log files are typically located in the Elasticsearch home folder or the `logs` subdirectory.\n"
+        "2. Run the following command to set the correct permissions for the log files:\n"
+        "      chmod 600 <Path_to_Log_Files>\n"
+        "3. Verify that the permissions have been set correctly by running the following command:\n"
+        "      ls -l <Path_to_Log_Files>\n"
+        "   The command should display the permissions as `-rw-------` for the Elasticsearch log files."  
+      cis_benchmark: {
+        profile_level: 1
+        severity: LOW
+      }
+      scan_instructions:
+        # /usr/share/elasticsearch
+        "generic{check_alternatives:{"
+        "  file_checks:{"
+        "    files_to_check:{single_file:{path:\"/usr/share/elasticsearch/elasticsearch.yml\"}}"
+        "    existence:{should_exist: true}"
+        "    non_compliance_msg: \"Elasticsearch configuration file does not exist\""
```

Comment:
For the places where we do leave in the existence_check I'd remove this non_compliance_msg as otherwise the user who gets the reports doesn't know exactly which file is missing.